arrow-core-test/src/main/kotlin/arrow/core/test/generators/Generators.kt

```diff
@@ -178,21 +178,7 @@ fun <T> Gen.Companion.id(gen: Gen<T>): Gen<Id<T>> = object : Gen<Id<T>> {
 }
 
 fun <A, B> Gen.Companion.ior(genA: Gen<A>, genB: Gen<B>): Gen<Ior<A, B>> =
-  object : Gen<Ior<A, B>> {
-    override fun constants(): Iterable<Ior<A, B>> =
-      (genA.orNull().constants().asSequence().k() to genB.orNull().constants().asSequence().k()).let { (ls, rs) ->
-        SequenceK.apply().run { ls.product(rs) }.filterMap {
-          Ior.fromOptions(Option.fromNullable(it.a), Option.fromNullable(it.b))
-        }.asIterable()
-      }
-
-    override fun random(): Sequence<Ior<A, B>> =
-      (Gen.option(genA).random() to Gen.option(genB).random()).let { (ls, rs) ->
-        ls.zip(rs).filterMap {
-          Ior.fromOptions(it.first, it.second)
-        }
-      }
-  }
+  genA.alignWith(genB) { it }
```

Comment:
We have it also used [here](https://github.com/arrow-kt/arrow-incubator/pull/109#discussion_r463748366) so, if this is merged then I'll update it to use this implementation. I also foresee more potential use cases, potentially existing ones.

Reply:
Also, cc @1Jajen1 :)

Reply:
Generators are alignable functors and this behavior is much preferable to the default in kotest which is zipping. 
This has use-cases for when we need 0-n values for example with product types that have defaults or sum types that have a None-like type.
When `constants` or `random` are infinite it will be indistinguishable from zip, but since `constants` rarely is, alignWith will be better.